Added documentation about autodetection for CMake bulild option. (#615)
geodynamics/Rayleigh: doc/source/User_Guide/getting_started.rst

Alternative: Configure using CMake
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> `CMake <https://cmake.org>`_ can be used as an alternative to the configure script. It is especially useful when running on a new platform not yet supported by configure or when you are generally more comfortable with CMake from other projects. It is recommended to load MKL library beforehand, and CMake can find it automatically. When using custom BLAS, LAPACK, and FFTW instead of MKL, loding them before running CMake command is recommended, as they will be automatically detected as well.

Reviewer: ```suggestion
`CMake <https://cmake.org>`_ can be used as an alternative to the configure script. It is especially useful when running on a new platform not yet supported by configure or when you are generally more comfortable with CMake from other projects. It is recommended to load the MKL or `cray-libsci` module beforehand, so CMake can find it automatically. When using custom BLAS, LAPACK, and FFTW instead of MKL, loading them before running `cmake` is recommended, as they will be automatically detected as well.
```

Slightly rewritten and fixed a typo. What do you think?

> `CMake <https://cmake.org>`_ can be used as an alternative to the configure script. It is especially useful when running on a new platform not yet supported by configure or when you are generally more comfortable with CMake from other projects. It is recommended to load the MKL or `cray-libsci` module beforehand, so CMake can find it automatically. When using custom BLAS, LAPACK, and FFTW instead of MKL, loading them before running `cmake` is recommended, as they will be automatically detected as well.

Reviewer: ```suggestion
`CMake <https://cmake.org>`_ can be used as an alternative to the configure script. It is especially useful when running on a new platform not yet supported by configure or when you are generally more comfortable with CMake from other projects. It is recommended to load the MKL or ``cray-libsci`` module beforehand, so CMake can find it automatically. When using custom BLAS, LAPACK, and FFTW instead of MKL, loading them before running ``cmake`` is recommended, as they will be automatically detected as well.
``

Sorry, adjusting to RST format.
